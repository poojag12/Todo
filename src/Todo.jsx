import React, {useState, useEffect} from 'react';
import Form from "./Form";

export function hello() {
    console.log("Hello World");
}

function Todo(){
    const [todos, setTodo] = useState([]);
    const [currentTodoText, addTodo] = useState("");

    useEffect(() => {
        hello();
    }, []);
    const addTodoClick = (e) => {
       const newTodo = [...todos, currentTodoText];
        e.preventDefault();
        setTodo(newTodo);
        addTodo("")
    };

    return (
        <div className="App">
            <h1>ToDo</h1>
            <div>
                <Form addTodo={addTodo} addTodoClick={addTodoClick}
                      currentValue={currentTodoText}/>
                <ul className="todo-list">{todos.map(todo => <h1 key={todo}>{todo}</h1>)}</ul>
            </div>
        </div>
    );
}

export default Todo;
