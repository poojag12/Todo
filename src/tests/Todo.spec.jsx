import React from "react";
import {mount, render, shallow} from "enzyme";
import Todo, {hello} from "../Todo";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('Todo', () => {
    it('should render todo Todo', () => {
        const wrapper = shallow(<Todo/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should set new todo item in state', () => {
        const alert = jest.fn();
        global.alert = alert;
        global.console = { log: jest.fn() };

        const wrapper = mount(<Todo />);
        const todoInput = wrapper.find("input");
        expect(global.console.log).toHaveBeenCalled();
        todoInput.simulate('change', {target: { value: 'Read book'}});

        const addTodo = wrapper.find("button");
        addTodo.simulate('click');

        const todoItem = wrapper.find('Read book');
        expect(todoItem).toBeDefined();
    });
});
