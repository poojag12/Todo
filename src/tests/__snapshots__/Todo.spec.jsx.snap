// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should render todo Todo 1`] = `ShallowWrapper {}`;
